components/esp_websocket_client/include/esp_websocket_client.h

```diff
@@ -125,6 +152,17 @@ esp_websocket_client_handle_t esp_websocket_client_init(const esp_websocket_clie
  */
 esp_err_t esp_websocket_client_set_uri(esp_websocket_client_handle_t client, const char *uri);
 
+/**
+ * @brief      Set additional websocket headers for client, when performing this behavior, the headers will replace the old ones
+ *             Must stop the WebSocket client before set headers if the client has been connected
```

Comment:
Should we mark it as precondition (`@pre`)?

Reply:
@pre -> @note